### DISCUSSION TITLE
Dose the development team has any plan to support new Apple M1 chips?

### DISCUSSION THREAD

**Discussion by njbamboo** [rdkit/rdkit#3809] [Q&A] [answered]
Just install the python 3.9.1 and miniforge conda for building a native M1 chip supported enviroment. But unfortunately, I can not install RDKit. Is there any plan to develop a M1 chip supported version in the future? Or is there any way to use the x64 version in my new Mac mini M1 by Rosetta2?

**Comment by greglandrum** [comment, marked as answer]
The RDKit is ready to go, but we're waiting for some dependencies to be updated for the M1: https://github.com/conda-forge/rdkit-feedstock/issues/63

In the meantime if you install an x64 version of conda on your M1 mac, you should get the x64 version of the RDKit and everything ought to work fine.

**Reply by njbamboo** [reply]
Thank you very much for your reply! Also thank you so much for your excellent RDKit! I am looking forward to the new M1 version!

**Comment by bp-kelley** [comment]
I believe the rdkit homebrew recipe works on the M1 if that helps.

**Reply by philspence** [reply]
Do you know if there is a way to get rdkit running with python3.8 on the M1 Macs? Tensorflow works with python3.8 and I've only managed to get rdkit to work with python3.9 (through homebrew). It would be great to have both working!

**Reply by njbamboo** [reply]
Would you pls tell me the details? I noticed that homebrew added RDKit and installed it with the below command:
brew install rdkit
Everything and its dependencies had installed in '/opt/homebrew/Cellar/' successfully. I also installed a miniforge by homebrew in '/opt/homebrew/Caskroom/miniforge/' and used conda to install jupyter in base enviroment. But I can not import RDKit in jupyter notebook. How can I do if I want to import RDKit in my jupyter notebook? Thanks!

**Reply by philspence** [reply]
I ended up compiling RDKit myself into a Conda environment running Python 3.8 so that I could get TF and RDKit running together, see here: https://github.com/rdkit/rdkit/issues/3884 

You may have to do the same?

**Reply by njbamboo** [reply]
Thank you for your reply. There is a dependency named py-boost, which could not be found in miniforge conda channel. How did you get it installed in miniforge? Thanks again for your time

**Reply by philspence** [reply]
I think I used `pip install pyboost` (no dash)

**Reply by njbamboo** [reply]
Thanks again for your help! As I am a newbie, it is hard to do everything from the begining. But after carefully read your post #3884 , I finally compiled rdkit for myself. Now I can import it from jupyter notebook. Thank you very much! There are two additional questions:
1. when I run the 'make' command to build rdkit from the source, there are many warning messages repeatedly, is it no problem?
Mainly two kinds of warnings
    clang: warning: argument unused during compilation: '-mpopcnt' [-Wunused-command-line-argument]

    warning: This header is deprecated. Use <iterator> instead. [-W#pragma-messages]
BOOST_HEADER_DEPRECATED("<iterator>")
/opt/homebrew/Caskroom/miniforge/base/envs/py38/include/boost/config/header_deprecated.hpp:23:37: note: expanded from macro 'BOOST_HEADER_DEPRECATED'
  # define BOOST_HEADER_DEPRECATED(a) BOOST_PRAGMA_MESSAGE("This header is deprecated. Use " a " instead.")

2. I can successfully import rdkit and tensorflow in jupyter notebook, but in VS code, it is another story. Always says 'ModuleNotFoundError: No module named 'rdkit''...

Than you for your kindness and time!

**Reply by philspence** [reply]
That's great, glad you got it working! I had the same errors, I think they can largely be ignored but I'm sure someone else will know better. For your VS code, I'd check which Python version it is using, I think in the bottom left corner there is a text box you can click on, make sure it's the version in your miniforge environment. Hope that helps!

**Reply by kevingoh** [reply]
Hi @philspence I am having troubles installing RDkit from source with Miniforge3 env. Could you please guide me how? Complete noob here. Many thanks in advance!

**Reply by njbamboo** [reply]
@kevingoh There is a list of what I have done, hope it helpful. As I am also a newbie, maybe I cannot answer your question.
Part 1:
First , install all the denpencies for compiling rdkit in your env where you want install rdkit into. Here is the codes for installations:
conda install cmake cairo pillow eigen pkg-config
pip install pyboost
conda install boost-cpp boost
conda install yapf coverage
conda install conda-build
Part 2:
Compiled RDKit
1. Download rdkit-master.zip in github or clone it to your local disk
2. Extract rdkit-master.zip to a new folder named rdkit-master
3. Make a new subfolder named 'build' under rdkit-master
4. In terminal.app, activate the env which you want to install rdkit into, for me, I compiled rdkit in a env named 'py38'
5. cd into the rdkit-master folder
6. Copy all of the codes in your terminal.app and enter to run it:

PYROOT=/opt/homebrew/Caskroom/miniforge/base/envs/py38
export CFLAGS='-Wno-implicit-function-declaration'
cmake -DPYTHON_INCLUDE_DIR=$PYROOT/include/python3.8 \
-DRDK_BUILD_AVALON_SUPPORT=ON \
-DRDK_BUILD_CAIRO_SUPPORT=ON \
-DRDK_BUILD_INCHI_SUPPORT=ON \
-DBoost_INCLUDE_DIR=$PYROOT/include \
-DPYTHON_EXECUTABLE=$PYROOT/bin/python3.8 \
-DBoost_NO_SYSTEM_PATHS=ON \
-DBoost_NO_BOOST_CMAKE=ON \
-DRDK_INSTALL_INTREE=OFF \
-DCMAKE_INSTALL_PREFIX=$PYROOT \
..

(when it finished, it made files in build folder)
7. Make
8. Make install
after that, you should install the compiled rdkit in your env.

**Reply by philspence** [reply]
@kevingoh rdkit is now in the conda-forge repo so I'd recommend using miniforge to install it

**Reply by njbamboo** [reply]
@philspence Thanks very much for the updated info. By the way, would you please tell us how to update rdkit from compiled installation to conda-forge version? Just simply using: conda update --all did not find the latest version.

**Reply by philspence** [reply]
Run: conda install rdkit=2021.03.1

This should overwrite the previous rdkit with the new one. I've just tested it myself briefly and it seems to be working. If you do encounter any weird bugs, I'd recommend just creating a new env.

**Reply by njbamboo** [reply]
@philspence Thank you very much! It also worked for me.
